src/amazons3/amazons3_client.bal

```diff
@@ -50,7 +50,13 @@ public type AmazonS3Client client object {
                         message = CLIENT_CREDENTIALS_VERIFICATION_ERROR_MSG, cause = verificationStatus);
             return clientConfigInitializationError;
         } else {
-            self.amazonS3 = new(baseURL, amazonS3Config.clientConfig);
+            http:ClientSecureSocket? clientSecureSocket = amazonS3Config?.secureSocketConfig;
+            if (clientSecureSocket is http:ClientSecureSocket) {
+                amazonS3Config.clientConfig.secureSocket = clientSecureSocket;
+                self.amazonS3  = new(baseURL, amazonS3Config.clientConfig);
+             } else {
+                self.amazonS3  = new(baseURL, amazonS3Config.clientConfig);
```

Comment:
We can take out _self.amazonS3  = new(baseURL, amazonS3Config.clientConfig);_ from if statement and remove the _else_ statement.

Reply:
Fixed